app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -368,9 +354,16 @@ def find_activation_keys
       activation_keys
     end
 
-    def find_foreman_host(organization)
-      Host.where(:name => params[:facts]['network.hostname'],
-                 :organization_id => organization.id).first if params[:facts]
+    def find_or_create_foreman_host(organization)
+      hosts = ::Host.where(:name => params[:facts]['network.hostname'])
+      if hosts.empty? #no host exists
+        ::Host::Managed.new_from_rhsm_params(system_params, organization, Location.default_location)
```

Comment:
Early in the review of this, but was surprised this wasn't somehow handled by the subscription aspect since RHSM is rather specific to that particular aspect.